src/MSAL.PCL/Internal/Cache/TokenCacheKey.cs

```diff
@@ -192,6 +173,24 @@ internal bool ScopeEquals(SortedSet<string> otherScope)
             return false;
         }
 
+        internal bool ScopeContains(SortedSet<string> otherScope)
+        {
+            foreach (string otherString in otherScope)
+            {
+                if (!Scope.Contains(otherString))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        public bool ScopeIntersects(SortedSet<string> otherScope)
+        {
+            return this.Scope.Intersect(otherScope).ToArray().Length > 0;
```

Comment:
I think you're looking for this: https://msdn.microsoft.com/en-us/library/dd411701(v=vs.110).aspx